Controller/SourceController.php

```diff
@@ -45,6 +45,12 @@ public function getAction($source) {
         $sourceConfig = $config['sources'][$source];
         /** @var AnalyticsInterface $analytics */
         $analytics = new $sourceConfig['class']($this->get('service_container'));
+        if (method_exists($analytics, 'setContext')) {
```

Comment:
You are right, is_callable is a better choice than method_exists.

To avoid the duck-typing, I'll use instanceof to make it cleaner.